src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -135,7 +134,52 @@ protected internal Resource(OperationSet operationSet, IEnumerable<Operation> op
 
         protected override IEnumerable<FieldDeclaration> GetAdditionalFields()
         {
+            // the resource data private field
             yield return new FieldDeclaration(FieldModifiers, ResourceData.Type, DataFieldName);
+
+            // the resource type public field
+            yield return new FieldDeclaration(
+                $"Gets the resource type for the operations",
+                FieldModifiers.Public | FieldModifiers.Static | FieldModifiers.ReadOnly,
+                typeof(ResourceType),
+                "ResourceType",
+                Literal(ResourceType.ToString()));
+        }
+
+        private IReadOnlyList<PropertyDeclaration>? _properties;
+        public IReadOnlyList<PropertyDeclaration> Properties => _properties ??= BuildProperties().ToArray();
+
+        protected virtual IEnumerable<PropertyDeclaration> BuildProperties()
```

Comment:
change return type to `IReadOnlyList<T>`? So that, there is no need of `ToArray`

Reply:
Never mind, the signature can't be changed.